### ISSUE TITLE
Notification delay

### ISSUE THREAD

**Issue by supervoidcoder** [ScratchAddons/ScratchAddons#7952]
### Description

I have noticed that the notifications ping on windows doesn't show up until I actually click the extension icon. The little indicator on the extension does update immediately, but it won't give me any push notifications until I click on it.

### Steps to reproduce

1.Get a message on scratch
2. See the icon change
3. No notification until clicked

### Environment

Browser: Firefox Version 132.0.2 (64-bit)
Scratch Addons version:  Version 1.39.2 Last Updated November 10, 2024
Operating system: Windows 10 Home 22H2

**Comment by iqnite** [comment]
This issue's been around for more than 1 year now, generally it can be said that `scratch-notifier` is not reliable at all.

**Comment by supervoidcoder** [comment]
> This issue's been around for more than 1 year now, generally it can be said that `scratch-notifier` is not reliable at all.

Yup, this has happened for months, or yeah more than a year. Do you think it's fixable though? Or is it something just wrong with Firefox? Does it work in the other browsers?

**Comment by Samq64** [comment]
> This issue's been around for more than 1 year now

So before May when we migrated to Mv3 in v1.38?

**Comment by supervoidcoder** [comment]
> > This issue's been around for more than 1 year now
> 
> So before May when we migrated to Mv3 in v1.38?

I was thinking that was the issue! I was looking through the scratch-notifier code and thought that it might have something to do with manifestv3. Apparently all the background scripts turned into service workers, which don't run persistently :(.
 Also, I'm no expert in JS (I'm still learning it) but I noticed you're using the 'chrome' instead of 'browser' so maybe that has something to do with the firefox compatibility?

**Comment by WorldLanguages** [comment]
> Also, I'm no expert in JS (I'm still learning it) but I noticed you're using the 'chrome' instead of 'browser' so maybe that has something to do with the firefox compatibility?

Pretty certain that's not it.

**Comment by supervoidcoder** [comment]
Huh, today I didn't even click the icon and I got a notification after months of it not working! Did it finally get fixed? 😀

**Comment by Samq64** [comment]
There was a patch release yesterday but it didn't touch Scratch notifier's code.

**Comment by supervoidcoder** [comment]
> There was a patch release yesterday but it didn't touch Scratch notifier's code.

Weird! Let me check if it wasn't a fluke.
